content/blueprints/gcp.md

```diff
@@ -0,0 +1,1277 @@
+
+{{< gsHighlight  yaml  >}}
+
+my_net:
+  type: cloudify.gcp.nodes.Network
+
+{{< /gsHighlight >}}
+
+
+
+
+
+## cloudify.gcp.nodes.Route
+**Derived From:** [cloudify.nodes.Router]({{< relref "blueprints/built-in-types.md" >}})
+
+A defined route, which will be added to the specified network.
+If tags are specified, it will only be added to Instances matching them.
+
+
+
+**Properties:**
+
+
+  * `dest_range`
+    The outgoing range that this route will handle
+
+    *required* None
+  * `priority`
+    The routing table priority for this route. Routes with lower priority numbers will be chosen first if more than one route with a matching prefix of the same length.
+
+    *default:* 1000
+  * `additional_settings`
+    Additional setting for firewall
+
+    *default:* {}
+  * `next_hop`
+    The Instance, IP or VpnTunnel which will handle the matching packets
+
+    *default:*
+  * `name`
+    Optional Route name. The instance ID will be used by default.
+
+    *default:*
+  * `tags`
+    Instance tags that this route will be applied to
+
+    *default:* []
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+
+    *default:* {}
+
+
+
+
+
+## cloudify.gcp.nodes.SecurityGroup
+**Derived From:** [cloudify.nodes.SecurityGroup]({{< relref "blueprints/built-in-types.md" >}})
+
+A virtual SecurityGroup.
+
+Google Cloud Platform has no entity equivalent to a Security Group on AWS or OpenStack, so as a convenience Cloudify includes a virtual one. It is implemented behind the scenes using a specially constructed tag and a number of FirewallRules.
+
+
+
+**Properties:**
+
+
+  * `rules`
+    List of FirewallRules which will form this SecurityGroup. Only the `sources:` and `allowed:` fields should be supplied (see FirewallRule properties for details).
+
+    *default:* []
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+
+    *default:* {}
+  * `name`
+    Optional security group name. By default it will be network name plus node name.
+
+    *default:*
+
+
+
+
+
+## cloudify.gcp.nodes.SslCertificate
+**Derived From:** [cloudify.nodes.Root]({{< relref "blueprints/built-in-types.md" >}})
+
+A TLS/SSL certificate and key. This will be used by a HTTPS TargetProxy to provide authenticated encryption for connecting users.
+
+
+
+**Properties:**
+
+
+  * `private_key`
+    Dictionary describing private key in PEM format used to generate this SSL certificate. Expected format is:
+      type: text|file
+      data: Private key in PEM format if text, otherwise path to a file with private key
+
+    *default:* {}
+  * `name`
+    Optional SSL certificate name. By default it will be SSL certificate id.
+
+    *type:* string
+    *default:*
+  * `certificate`
+    Certificate (self-signed or obtained from CA) in PEM format. Expected format is:
+      type: text|file
+      data: Certificate in PEM format if text, otherwise path to a file with certificate
```

Comment:
data: **The**  ~~C~~ **c**ertificate in PEM format if text, otherwise **the** path to **the** ~~a~~ file **containing the** ~~with~~ certificate **.**